lib/agent.js

```diff
@@ -22,6 +23,18 @@ function defaultFactory (origin, opts) {
     : new Pool(origin, opts)
 }
 
+function getDNSResolver (options) {
+  if (options.DNSResolver instanceof DNSResolver) {
+    return options.DNSResolver
+  }
+
+  if (options?.dnsResolver?.disable) {
+    return undefined
+  }
```

Comment:
If someone provides a DNSResolver option *and* a `dnsResolver` option, the former takes precedence. Having `disable: true` should take precedence instead. It might also be better off being renamed to dnsResolverOptions or something so there isn't any confusion.

Reply:
Done